pkgs/development/python-modules/mlx/default.nix

```diff
@@ -72,49 +66,69 @@ let
     # updates the wrong fetcher rev attribute
     passthru.skipBulkUpdate = true;
 
-    env = {
-      PYPI_RELEASE = version;
-      CMAKE_ARGS = toString [
-        # NOTE The `metal` command-line utility used to build the Metal kernels is not open-source.
-        # To build mlx with Metal support in Nix, you'd need to use one of the sandbox escape
-        # hatches which let you interact with a native install of Xcode, such as `composeXcodeWrapper`
-        # or by changing the upstream (e.g., https://github.com/zed-industries/zed/discussions/7016).
-        (lib.cmakeBool "MLX_BUILD_METAL" false)
-        (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_GGUFLIB" "${gguf-tools}")
-        (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_JSON" "${nlohmann_json.src}")
-        (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_FMT" "${fmt_10.src}")
-      ];
-    };
+    env.PYPI_RELEASE = "1";
+
+    env.CMAKE_ARGS =
+      let
+        gguf-tools-src = fetchFromGitHub {
+          owner = "antirez";
+          repo = "gguf-tools";
+          rev = "8fa6eb65236618e28fd7710a0fba565f7faa1848";
+          hash = "sha256-15FvyPOFqTOr5vdWQoPnZz+mYH919++EtghjozDlnSA=";
+        };
+      in
+      toString (
+        [
+          # NOTE The `metal` command-line utility used to build the Metal kernels is not open-source.
+          # To build mlx with Metal support in Nix, you'd need to use one of the sandbox escape
+          # hatches which let you interact with a native install of Xcode, such as `composeXcodeWrapper`
+          # or by changing the upstream (e.g., https://github.com/zed-industries/zed/discussions/7016).
+          (lib.cmakeBool "MLX_BUILD_METAL" false)
+          (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_GGUFLIB" "${gguf-tools-src}")
+          (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_JSON" "${nlohmann_json.src}")
+          (lib.cmakeOptionType "filepath" "FETCHCONTENT_SOURCE_DIR_FMT" "${fmt_10.src}")
+        ]
+        ++ lib.optionals stdenv.hostPlatform.isLinux [
+          # RPATH of binary contains a forbidden reference to /build/
+          (lib.cmakeBool "CMAKE_SKIP_BUILD_RPATH" true)
+          # the default is lib64, which is not in the RPATH of the core binary
+          (lib.cmakeFeature "CMAKE_INSTALL_LIBDIR" "lib")
+        ]
+      );
 
     build-system = [
-      setuptools
-    ];
-
-    nativeBuildInputs = [
       cmake
```

Comment:
the `cmake` used here is not `pkgs.cmake` but `python3Packages.cmake` so it'd make sense to put it into `build-system` (where python build-time deps go)